### PR TITLE
feat: add `beforeRender` option

### PR DESCRIPTION
**Pull request by arturovt**
Hey, I haven't found any best way to implement it, so I've decided to add a `beforeRender` option. I've left an example in the readme on what goal is trying to be accomplished.

I'm open to your comments.

### CODE REVIEW

File: projects/ngx-quill/src/lib/quill-editor.component.ts [hunk 1]
```diff
@@ -220,7 +221,16 @@ export abstract class QuillEditorBase implements AfterViewInit, ControlValueAcce
     // this will lead to runtime exceptions, since the code will be executed on DOM nodes that don't exist within the tree.
 
     this.quillSubscription = this.service.getQuill().pipe(
-      mergeMap(Quill => this.service.registerCustomModules(Quill, this.customModules))
+      mergeMap((Quill) => {
+        const promises = [this.service.registerCustomModules(Quill, this.customModules)]
+        if (this.beforeRender) {
+          promises.push(this.beforeRender())
+        }
+        if (this.service.config.beforeRender) {
```
Comment (KillerCodeMonkey) [inline projects/ngx-quill/src/lib/quill-editor.component.ts:229]: i would make the "beforeRender" at the component overriding the service.beforeRender.

All inputs and configs are currently working this way. What do you think?

Comment (KillerCodeMonkey) [reply projects/ngx-quill/src/lib/quill-editor.component.ts:229]: and tests would be nice :)

**Review by KillerCodeMonkey — CHANGES_REQUESTED** [review]

**Comment by arturovt** [comment]
Addressed comments, added tests.

**Comment by arturovt** [comment]
Yeah, rebased on master, thank you.

**Review by KillerCodeMonkey — APPROVED** [review]

**Pull request merged**